client/my-sites/themes/theme-showcase.jsx

```diff
@@ -378,9 +379,20 @@ class ThemeShowcase extends Component {
 			...( isCollectionView && tier && ! filter && { tabFilter: '' } ),
 		};
 
+		const showCollections = tier === '' && this.isThemeDiscoveryEnabled() && ! isCollectionView;
```

Comment:
The collection should only be displayed in Recommended/All, so we need to also check `filter` here.

Reply:
See https://github.com/Automattic/wp-calypso/pull/83814#discussion_r1381316496

Reply:
Are we sure that we're going to display the collection on the `All` tab? It would be the first time we do so and it's not specified anywhere that it should work like this.

Reply:
While we discuss including the `All` tab I've included the collection carousel in it and changed this condition a bit more. This is how it works now:
- Shown on `Recommended` & `All` tabs.
- No search must be specified.
- No filter must be specified.
- No tier must be specified.

Reply:
Recommended/All not All :)

<img width="977" alt="Screenshot 2023-11-03 at 09 46 37" src="https://github.com/Automattic/wp-calypso/assets/2070010/b3ca21bb-a920-4afb-887b-7d80af885d28">

Reply:
(Apologies for being very unclear on that)

Reply:
Done.